lib/order.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'menu'
+require_relative 'sms'
+
+class Order
+  attr_accessor :order_array, :menu
+  ORDER_MIN = 1
+
+  def initialize
```

Comment:
So you have two dependent classes? 'Menu' and 'SMS'. If you want to get your doubles working for your test, you'll need to pass these as optional default parameters to the initialize method.

i.e.

`def initialize(menu_class = Menu, sms_class = SMS)
    @order_array = []
    @menu = menu_class.new
    @sms = sms_class.new
end
'

This will enable you to pass the class_double as a parameter to the target class in the rspec, overiding the actual class dependency.